modules/ask_va_api/app/lib/ask_va_api/inquiries/payload_builder/inquiry_details.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+module AskVAApi
+  module Inquiries
+    module PayloadBuilder
+      class InquiryDetails
+        include SharedHelpers
+
+        attr_reader :inquiry_params
+
+        def initialize(inquiry_params)
+          @inquiry_params = inquiry_params
+        end
+
+        def call
+          inquiry_details = base_inquiry_details(inquiry_params[:your_role])
+
+          if education_benefits?(inquiry_params[:select_category],
+                                 inquiry_params[:select_topic])
+            return general_inquiry(inquiry_params, inquiry_details)
+          end
+
+          if inquiry_params[:who_is_your_question_about] == 'Myself'
+            return handle_self_inquiry(inquiry_params, inquiry_details)
+          end
+
+          if inquiry_params[:who_is_your_question_about] == 'Someone else'
+            handle_others_inquiry(inquiry_params, inquiry_details)
+          end
+        end
```

Comment:
I'm not sure how you plan to implement this class, but if you wanted to further refactor this consider:

splitting the `call` method into three public methods: `general_inquiry`, `self_inquiry`, `other_inquiry` and perform the if statement checks in the class that calls `InquiryDetails#call`.

Or you could make `InquiryDetails` the super class and subclass `GeneralInquiries`, `SelfInquiries`, `OtherInquiries` each with the call method then you wouldn't have to use as many if statements in the private methods.